clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitJobExecutor.java

```diff
@@ -75,6 +76,29 @@ public GitJobExecutor(
     askPassBinary = initAskPass();
   }
 
+  public void cloneOrPull(String repoUrl, String branch, Path localPath, String repoBasename)
+      throws IOException {
+    if (!Files.exists(localPath)
+        || localPath.toFile().listFiles() == null
+        || localPath.toFile().listFiles().length == 0) {
+      clone(repoUrl, branch, localPath);
+      return;
+    }
+    if (!Files.isDirectory(localPath)) {
+      throw new IllegalArgumentException(
+          "Local path " + localPath.toString() + " is not a directory");
+    }
+    Path clonePath = Paths.get(localPath.toString(), repoBasename);
+    File[] files = clonePath.toFile().listFiles();
+    if (files == null
+        || files.length == 0
+        || Arrays.stream(files).noneMatch(f -> f.getName().equals(".git"))) {
```

Comment:
The files are ignored so just testing on `clonePath.resolve(".git").toFile().exists()` without listing the files be better. At the same time, if `clonePath` has files shouldn't we give an exception and fallback on pulling?

Reply:
If `clonePath` has `.git` we can pull, if not we fallback to clone, which nukes the directory to start fresh again. 
Since we already tested that `localPath` has files above, I'll add a warning in that case indicating that the directory has files but there's no `.git`, so they will be deleted to start with a fresh clone.